### DISCUSSION TITLE
Date handling, new MicroGui widget

### DISCUSSION THREAD

**Discussion by peterhinch** [micropython/micropython#10896] [Show and tell]
I developed these for a project of my own, but they are designed to be of general use:
 1. [A DS3231 driver](https://github.com/peterhinch/micropython-samples/blob/master/DS3231/README.md). The chip is a very cheap and highly accurate RTC. The driver supports alarms and is much more compact than other solutions (the test script is longer than the driver).
 2. [A Date class](https://github.com/peterhinch/micropython-samples/blob/master/date/DATE.md) The official `datetime` class is more full-featured but it is quite large. This "micro" solution has no concept of time but enables date calculations and calendar support.
 3. [A grid widget](https://github.com/peterhinch/micropython-micro-gui/blob/main/README.md#611-grid-widget) for MicroGui (image below). This supports spreadsheet-like displays of arbitrary data.

![Image](https://github.com/peterhinch/micropython-micro-gui/raw/main/images/grid.JPG)

**Comment by beetlegigg** [comment]
Nice one  @peterhinch,  I think that will go nicely on one of my ili9341 screens that I connect to a rpi pico, especially as I've been 3d printing some nice cases for that setup that even my wife approves of.  I will see if I can also squeeze your label widget onto the same screen for a clock display  too.   Marvellous.
